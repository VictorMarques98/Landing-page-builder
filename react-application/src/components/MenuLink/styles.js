/* eslint-disable no-unused-vars */
import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.font.sizes.xsmall};
    padding: ${theme.font.spacings.xsmall};
    color: ${theme.colors.primaryColor};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      left: 50%;
      transition: all 300ms ease-in-out;
    }

    &:hover::after {
      width: 50%;
      left: 25%;
    }
  `}
`;
