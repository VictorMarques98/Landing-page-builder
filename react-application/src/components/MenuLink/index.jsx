import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './styles';

export const MenuLink = ({ children, link, newTab }) => {
  const target = newTab ? '_blank' : '_self';
  return (
    <Styled.Container href={link} target={target}>
      {children}
    </Styled.Container>
  );
};

export default MenuLink;

MenuLink.defaultProps = {
  children: {},
  link: '',
  newTab: false,
};

MenuLink.propTypes = {
  children: PropTypes.node.isRequired,
  link: PropTypes.string.isRequired,
  newTab: PropTypes.bool,
};
