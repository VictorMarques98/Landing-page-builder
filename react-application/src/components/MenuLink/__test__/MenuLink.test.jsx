import { screen } from '@testing-library/react';
import React from 'react';
import MenuLink from '../';
import renderTheme from '../../../styles/render-theme';

describe('<MenuLink />', () => {
  it('should render a link', () => {
    renderTheme(<MenuLink link="https://www.google.com/">Children</MenuLink>);
    const menuLink = screen.getByRole('link', { name: 'Children' });
    expect(menuLink).toBeInTheDocument();
    expect(menuLink).toHaveAttribute('target', '_self');
  });

  it('should open in a new tab', () => {
    renderTheme(
      <MenuLink link="https://www.google.com/" newTab={true}>
        Children
      </MenuLink>,
    );
    const menuLink = screen.getByRole('link', { name: 'Children' });
    expect(menuLink).toHaveAttribute('target', '_blank');
  });

  it('should oto match snapshot', () => {
    renderTheme(
      <MenuLink link="https://www.google.com/" newTab={true}>
        Children
      </MenuLink>,
    );
    const { container } = screen.getByRole('link', { name: 'Children' });
    expect(container).toMatchSnapshot();
  });
});
