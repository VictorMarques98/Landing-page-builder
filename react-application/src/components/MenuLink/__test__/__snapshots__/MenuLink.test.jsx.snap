// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match snapshot 1`] = `
.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.76rem;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  left: 50%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

<a
  class="c0"
  href="https://www.google.com/"
  target="_blank"
>
  Children
</a>
`;
